src/ImageObject.py

```diff
@@ -5,37 +5,106 @@
 #               be randomly varied.
 #
 
+from texture import salt_and_pepper, random_uniform, AdvancedTextures
+
 import random
 import numpy as np
 
 from abc import ABC, abstractmethod
 from OpenGL.GL import *
 
+import pygame
+
 
 class ImageObject(ABC):
     """
     An object that gets rendered onto an image. The object will have a general shape, but will
     have randomly generated qualities on each construction OR when commanded. Keeps track 
     and highlights the verticies for corner detection.
     """
+    TEXTURE_DIM = (300, 300)
     VERTEX_COLOR = (0, 1, 0)
+    POS_WIDTH_LIMIT = 1
+    POS_HEIGHT_LIMIT = 0.5
+    ROT_LIMIT = 360
 
-    def __init__(self):
+    def __init__(self, base_color):
+        # color information
+        self.base_color = base_color
         self.color = None
-        self.vertices = None
-        self.pos = None
-        self.rot = None
+        # make texture
+        self.texture = None
+        self.texture_data = None
+        self.texture_id = None
+        self.has_texture = False
+        self.tex_coords = []
+        # object shape 
+        self.vertices = []
+        self.normals = []
+        # object orientation
+        self.pos = [] # x y z
+        self.rot = [] # angle_x angle_y angle_z
+        # setting object properties
+        self.generate_color()
+        self.generate_position()
+        self.generate_texture()
 
-    @abstractmethod
-    def generate_shape(self, seed=None):
-        """
-        Randomly picks out the verticies to produce the shape in question. This function
-        will fill in the vertex arrays, which will be used to render in the render step.
+    def __del__(self):
+        if self.texture_id is not None:
+            glDeleteTextures(self.texture_id)
 
-        Inputs:
-            seed (int): the seed for the random number generator
-        """
-        pass
+    def generate_texture(self, wantTexture=False):
+        # 50% change of texture
+        if not wantTexture:
+            if random.random() > 0.5:
+                self.has_texture = False
+                return
+        
+        self.has_texture = True
+        # roll the dice to see what texture to use
+        texture_idx = random.randint(1, 5)
+        # loading texture image
+        self.texture = pygame.image.load("textures/texture{}.png".format(texture_idx))
```

Comment:
only works when running from src/ folder. 
should replace path with `os.path.join(os.path.dirname(__file__), "textures/texture{}.png".format(texture_idx)`